src/main/java/uk/gov/pay/products/resources/ProductMetadataResource.java

```diff
@@ -44,11 +42,33 @@ public Response createMetadata(@PathParam("productExternalId") String productExt
 
         List<ProductMetadata> metadataList = metadataFactory.metadataFinder().findMetadataByProductExternalId(productExternalId);
 
-        return requestValidator.validateCreateRequest(payload, metadataList)
+        return requestValidator.validateRequest(payload)
                 .map(errors -> Response.status(Response.Status.BAD_REQUEST).entity(errors).build())
-                .orElseGet(() -> {
-                    metadataFactory.metadataCreator().createProductMetadata(payload, productExternalId);
-                    return Response.status(Response.Status.CREATED).build();
-                });
+                .orElseGet(() ->
+                    requestValidator.validateCreateRequest(payload, metadataList)
+                            .map(errors -> Response.status(Response.Status.BAD_REQUEST).entity(errors).build())
+                            .orElseGet(() -> {
+                                metadataFactory.metadataCreator().createProductMetadata(payload, productExternalId);
+                                return Response.status(Response.Status.CREATED).build();
+                            }));
+    }
+
+    @PATCH
+    @Path("/products/{productExternalId}/metadata")
+    @Produces(APPLICATION_JSON)
```

Comment:
I think this is the same for all metadata routes - could it be applied to the resource instead of individual requests?

Reply:
👍